### PR TITLE
docs: create local development docs

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Create local development docs based off [renovatebot/renovate/docs/development/local-development.md](https://github.com/renovatebot/renovate/blob/master/docs/development/local-development.md)
- Adjust docs where needed so that they have the correct instructions

## Context:

We have nothing that accurately describes the current steps to re-create the documentation website locally.
I know we're going to migrate to Docusaurus v2 sometime, but we should have at least a template that we can fill in properly later.

### CODE REVIEW

File: .github/local-development.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+curl -sL https://deb.nodesource.com/setup_12.x | sudo -E bash -
+curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
+echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
+sudo apt-get update
+sudo apt-get install -y git python-minimal build-essential nodejs yarn
+```
+
+_Windows_
+
+QUESTION: Git Bash for Windows does not come with `make` by default, how should a user work around this? Or is Windows not supported for making the docs locally?
```
Comment (HonkingGoose) [inline .github/local-development.md:35]: Do we even support Windows for the current build process?
The GitHub Actions use `ubuntu-latest`.

File: .github/local-development.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+For example, if you think anything is unclear, or you think something needs to be added, open a pull request!
+
+## Installation
+
+### Prerequisites
+
+You need the following dependencies for local development:
+
+- Git
+- Node.js `^12.13.0 || >=14.15.0`
```
Comment (HonkingGoose) [inline .github/local-development.md:14]: Maybe we should use:

```suggestion
- Node.js `^12.13.0`
```

As that is what the GitHub action is using now.

Have we tested that Node 14 will work?

File: .github/local-development.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+- Yarn `^1.17.0`
+- Python `^3.8`
```
Comment (HonkingGoose) [inline .github/local-development.md:16]:
```suggestion
- Python `^3.9`
```

Do we want to use the latest stable Python release?

File: .github/local-development.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+- Bash shell with `make`
+
+We support Node.js versions according to the [Node.js release schedule](https://github.com/nodejs/Release#release-schedule).
+
+_Linux_
+
+You can use the following commands on Ubuntu.
+
+```sh
+curl -sL https://deb.nodesource.com/setup_12.x | sudo -E bash -
```
Comment (HonkingGoose) [inline .github/local-development.md:26]:
```suggestion
curl -sL https://deb.nodesource.com/setup_14.x | sudo -E bash -
```

This was pointing to Node 14 on the Renovate docs, but I pointed it to 12 here.

File: .github/local-development.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+If you already installed a component, skip the corresponding step.
+
+- Install [Git](https://git-scm.com/downloads). Make sure you've [configured your username and email](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup)
+- Install [Node.js LTS](https://nodejs.org/en/download/)
+- In an Administrator PowerShell prompt, run `npm install -global npm` and then `npm --add-python-to-path='true' --debug install --global windows-build-tools`
+- Install [Yarn](https://yarnpkg.com/lang/en/docs/install/#windows-stable)
+
+  Verify you're using the appropriate versions:
+
+  ```powershell
```
Comment (HonkingGoose) [inline .github/local-development.md:47]: I don't know if you like the syntax highlighting for the PowerShell terminal here or not, feel free to tell me to remove it again. 😄

File: .github/local-development.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+If you want to contribute to the project, you should first "fork" the main project using the GitHub Website and then clone your fork locally.
+The Renovate project uses the [Yarn](https://github.com/yarnpkg/yarn) package management system instead of npm.
+
+To ensure everything is working properly on your end, you must:
+
+1. Make sure you don't have a local `.npmrc` file that overrides npm's default registry
+1. Install all npm dependencies with `yarn install`
+1. Install all Python dependencies with `pip install -r requirements.txt`
+1. Make a build with `make`
+1. Serve the build of the website locally with `mkdocs serve`
```
Comment (HonkingGoose) [inline .github/local-development.md:69]: Please check on your end if this is accurate to how thing work now.

File: .github/local-development.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+You only need to do these 5 steps this one time.
+
+Before you submit a pull request you should:
+
+1. Install newer npm dependencies with `yarn install`
+1. Install newer Python dependencies with `pip install -r requirements.txt`
+1. Make a build with `make`
+1. Serve your build locally with `mkdocs serve` and inspect if everything is working
```
Comment (HonkingGoose) [inline .github/local-development.md:78]: Again: please check if this is accurate.

File: .github/local-development.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+<!-- TODO: setup snapshot tests with Jest on React components. -->
+
+### Coverage
+
+<!-- TODO: fill in once we have tests and a coverage bot -->
+
+## Linting and formatting
+
+<!--- TODO: Once Prettier is setup, explain how to use it. -->
```
Comment (HonkingGoose) [inline .github/local-development.md:96]: This whole section is basically placeholder for when we do have those things.

File: .github/local-development.md (hunk 1 above)
Comment (viceice) [reply .github/local-development.md:35]: Nope, i think windows users should use wsl.

File: .github/local-development.md (hunk 3 above)
Comment (viceice) [reply .github/local-development.md:16]: Sure latest

File: .github/local-development.md (hunk 2 above)
Comment (viceice) [reply .github/local-development.md:14]: We only support latest lts. We should fix the action

File: .github/local-development.md (hunk 1 above)
Comment (HonkingGoose) [reply .github/local-development.md:35]: Okay will rewrite to point them to Windows Subsystem for Linux then.

Do we want them to use WSL1 or WSL2?

File: .github/local-development.md (hunk 2 above)
Comment (HonkingGoose) [reply .github/local-development.md:14]: I will open a PR to fix the action then. 👍

File: .github/local-development.md (hunk 5 above)
Comment (HonkingGoose) [reply .github/local-development.md]: Ignore this comment, as this is no longer in the docs, we now point to WSL2 instead. 😄

**Review by viceice — CHANGES_REQUESTED** [review]
I think we should defer this until docusaurus. Trying to get it done soon

**Comment by HonkingGoose** [comment]
@viceice Do you want to keep this branch around? Or do you want to close the PR and start fresh when you have Docusaurus v2 ready?

**Comment by viceice** [comment]
Whatever you prefer. 🤗

**Comment by HonkingGoose** [comment]
@viceice I'll keep the branch and PR alive then, we can keep some parts when we're using Docusaurus v2. 😄

**Comment by HonkingGoose** [comment]
I still think we'd benefit from having the current instructions in there.

I've updated the instructions, please confirm they are correct.

**Review by HonkingGoose — COMMENTED** [review]
I've just tested the getting started instructions, and it works as described on my end. 😄

**Review by viceice — APPROVED** [review]
LGTM but will change in near future. 😉

**Comment by HonkingGoose** [comment]
> LGTM but will change in near future. 😉

Okay! 😄

**Pull request merged**